browser/static/js/adjust.js

```diff
@@ -384,3 +343,97 @@ class ImageAdjuster {
     }
   }
 }
+
+class ChangeContrast extends Action {
+  constructor(adjuster, change) {
+    super();
+    this.adjuster = adjuster;
+    this.oldValue = adjuster.contrast;
+
+    const modContrast = -Math.sign(change) * 4;
+    // stop if fully desaturated
+    let newContrast = Math.max(adjuster.contrast + modContrast, adjuster.minContrast);
+    // stop at 8x contrast
+    newContrast = Math.min(newContrast, adjuster.maxContrast);
+    this.newValue = newContrast;
+  }
+
+  do() { this.setContrast(this.newValue); }
+
+  undo() { this.setContrast(this.oldValue); }
+
+  redo() { this.do(); }
+
+  setContrast(contrast) {
+    this.adjuster.rawLoaded = false;
+    this.adjuster.contrast = contrast;
+    this.adjuster.contrastRaw();
+  }
+}
+
+class ChangeBrightness extends Action {
+  constructor(adjuster, change) {
+    super();
+    this.adjuster = adjuster;
+    const modBrightness = -Math.sign(change);
+    // limit how dim image can go
+    let newBrightness = Math.max(adjuster.brightness + modBrightness, adjuster.minBrightness);
+    // limit how bright image can go
+    newBrightness = Math.min(newBrightness, adjuster.maxBrightness);
+    this.oldValue = adjuster.brightness;
+    this.newValue = newBrightness;
+  }
+
+  do() { this.setBrightness(this.newValue); }
+
+  undo() { this.setBrightness(this.oldValue); }
+
+  redo() { this.do(); }
+
+  setBrightness(brightness) {
+    this.adjuster.rawLoaded = false;
+    this.adjuster.brightness = brightness;
+    this.adjuster.contrastRaw();
+  }
+}
+
+class ResetBrightnessContrast extends Action {
+  constructor(adjuster) {
+    super();
+    this.adjuster = adjuster;
+    this.brightness = adjuster.brightness;
+    this.contrast = adjuster.contrast;
+  }
+
+  do() {
+    adjuster.brightness = 0;
+    adjuster.contrast = 0;
+    adjuster.rawLoaded = false;
+    adjuster.contrastRaw();
+  }
+
+  undo() {
+    adjuster.brightness = this.brightness;
+    adjuster.contrast = this.brightness;
+    adjuster.rawLoaded = false;
+    adjuster.contrastRaw();
```

Comment:
```suggestion
    this.adjuster.brightness = this.brightness;
    this.adjuster.contrast = this.contrast;
    this.adjuster.rawLoaded = false;
    this.adjuster.contrastRaw();
```
fixes the unexpected undo behavior I was seeing in user testing